### ISSUE TITLE
Cache to avoid reformatting doesn't work in v0.54.0 or later

### ISSUE THREAD

**Issue by pixyzehn** [nicklockwood/SwiftFormat#1768]
Hi!

First of all, thank you for maintaining such a great tool. I would like to report an issue where the cache to avoid reformatting doesn't seem to work anymore in v0.54.0 or later (v0.54.1).

For example, if I run `swiftformat .` twice in a row (without any changes), I got the following in each version:

- v0.54.0 or v0.54.1: `SwiftFormat completed in 3.99s` with cache

```sh
$ swiftformat .
Running SwiftFormat...
Reading config file at /[project path]/.swiftformat
Reading swift-version file at /[project path]/.swift-version (version 5.10)
SwiftFormat completed in 3.94s.
0/2200 files formatted, 5 files skipped.

$ swiftformat .
Running SwiftFormat...
Reading config file at /[project path]/.swiftformat
Reading swift-version file at /[project path]/.swift-version (version 5.10)
SwiftFormat completed in 3.99s.
0/2200 files formatted, 5 files skipped.
```

- v0.53.10: `SwiftFormat completed in 0.66s` with cache

```sh

$ swiftformat .
Running SwiftFormat...
Reading config file at /[project path]/.swiftformat
Reading swift-version file at /[project path]/.swift-version (version 5.10)
SwiftFormat completed in 3.94s.
0/2200 files formatted, 5 files skipped.

$ swiftformat .
Running SwiftFormat...
Reading config file at /[project path]/.swiftformat
Reading swift-version file at /[project path]/.swift-version (version 5.10)
SwiftFormat completed in 0.66s.
0/2200 files formatted, 5 files skipped.
```

Do you see any potential issues between https://github.com/nicklockwood/SwiftFormat/compare/0.53.10...0.54.0? Please let me know if you need more information such as `.swiftformat` file.

**Comment by nicklockwood** [comment]
@pixyzehn thanks for flagging this! It was indeed broken. I've fixed it now and added a test to prevent it regressing again.

**Comment by pixyzehn** [comment]
@nicklockwood Thanks for taking care of this so quickly!

**Comment by nicklockwood** [comment]
@pixyzehn fixed in 0.54.2
